core/src/main/java/org/renjin/primitives/combine/RowBindFunction.java

```diff
@@ -0,0 +1,157 @@
+package org.renjin.primitives.combine;
+
+import com.google.common.collect.Lists;
+import com.google.common.collect.Maps;
+import org.renjin.eval.Context;
+import org.renjin.eval.EvalException;
+import org.renjin.invoke.annotations.ArgumentList;
+import org.renjin.invoke.annotations.Current;
+import org.renjin.invoke.codegen.ArgumentIterator;
+import org.renjin.sexp.*;
+
+import java.util.List;
+import java.util.Map;
+
+/**
+ * Special function to do rbind and use objectnames as rownames
+ */
+public class RowBindFunction extends AbstractBindFunction {
+
+  public RowBindFunction() {
+    super("rbind");
+  }
+
+  @Override
+  public SEXP apply(Context context, Environment rho, FunctionCall call, PairList arguments) {
+
+    ArgumentIterator argumentItr = new ArgumentIterator(context, rho, arguments);
```

Comment:
The argument evaluation code is duplicated between rbind/cbind. Extract to a common method in `AbstractBindFunction.`